src/pages/dataSetNotificationTemplates/Edit.tsx

```diff
@@ -0,0 +1,217 @@
+import { useAlert, useDataEngine } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { Button, CircularLoader, NoticeBox } from '@dhis2/ui'
+import { useQuery } from '@tanstack/react-query'
+import React, { useMemo } from 'react'
+import { Form } from 'react-final-form'
+import { useParams } from 'react-router-dom'
+import {
+    DefaultFormFooter,
+    DefaultSectionedFormSidebar,
+    SectionedFormLayout,
+} from '../../components'
+import { SectionedFormProvider, DEFAULT_FIELD_FILTERS } from '../../lib'
+import { DataSetNotificationsFormFields } from './form/DataSetNotificationsFormFields'
+import {
+    getInitialValuesFromTemplate,
+    DataSetNotificationTemplate,
+    DataSetNotificationFormValues,
+    transformFormValues,
+} from './form/getInitialValuesFromTemplate'
+import { useOnEditNotifications } from './form/useOnEditNotifications'
+
+type Params = {
+    id: string
+}
+
+const fieldFilters = [
+    ...DEFAULT_FIELD_FILTERS,
+    'name',
+    'code',
+    'subjectTemplate',
+    'messageTemplate',
+    'notificationRecipient',
+    'dataSetNotificationTrigger',
+    'relativeScheduledDays',
+    'sendStrategy',
+    'recipientUserGroup',
+    'dataSets',
```

Comment:
You can fetch the `dataSet`-info in the same request instead in a separate request, using field-filters:
`dataSets[id,name,displayName]`. Then you should be able to remove the fetching of dataSets below.